lib/solidus_subscriptions.rb

```diff
@@ -2,7 +2,7 @@
 
 require 'solidus_core'
 require 'solidus_support'
-
+require 'solidus_legacy_promotions'
```

Comment:
Why do we need this?

Reply:
Without this, I am getting error `uninitialized constant Spree::PromotionHandler` hence I have added.

Reply:
But that class is never called in this extension. I suspect the problem is somewhere else. Do you have a full backtrace of the error?

Reply:
You are right, I don't know what I was looking for. 😓  Also this extension adds some promotion rules for the legacy system. The quick way is to add legacy_promtionm as a dependency of this extension via its gemfile.  But I guess the right way is to determine which promotion system has been used and call the right code based on that. @mamhoff if you have any guidance here, it would be appreciated.

Reply:
@kennyadsl let's keep in legacy promotion and fix along the way. What is the ETA for full deprecation with 5.0?

Reply:
@shahmayur001 given what @kennyadsl wrote we will restore functionality using the old promotion system. @mamhoff we would really appreciate your input here as the changes you have made are quiet hefty and complex. 

@shahmayur001 as soon as we have the basic set of extensions running we will revisit the changes to be made to have it all future proofed. It's neither a priority for us nor for Solidus to have this running with the old promotion gem removed.

Reply:
I've looked through both uses of the Promotion Handler here. You can just remove them and everything should work as it should.

Reply:
thanks @mamhoff! 

@shahmayur001 you should be able to understand if there's the new or the legacy promotion system easily. Based on that we can avoid calling the code that only works with legacy.

Reply:
> I've looked through both uses of the Promotion Handler here. You can just remove them and everything should work as it should.

what happens in that case if a promotion is applied to a product?

Reply:
1. The extension should work with both promo systems. You can check for the existence of the old promotion system by checking `Object.const_defined?("Spree::Promotion")`. 
2. Yes.

Reply:
We are good than. @shahmayur001 great work @mamhoff thank you for your help!